hyperspy/_signals/eds_tem.py

```diff
@@ -325,12 +331,36 @@ def quantification(self,
             navigation_mask = navigation_mask.data
         xray_lines = self.metadata.Sample.xray_lines
         composition = utils.stack(intensities)
-        composition.data = utils_eds.quantification_cliff_lorimer(
-            composition.data, kfactors=kfactors,
-            mask=navigation_mask) * 100.
+        if method == 'CL':
+            composition.data = utils_eds.quantification_cliff_lorimer(
+                composition.data, kfactors=factors,
+                mask=navigation_mask) * 100.
+        elif method == 'zeta':
+            results = utils_eds.quantification_zeta_factor(
+                composition.data, zfactors=factors,
+                dose=self._get_dose(method))
+            composition.data = results[0] * 100.
+            mass_thickness = intensities[0]
```

Comment:
Should be `mass_thickness = intensities[0].deepcopy()`, if not, intensity[0] will be overwritten in the next line (as mass_thickness points to the same array as intensities[0] in the current code).